src/Api/Helpers/AddDatabaseExtensionMethod.cs

```diff
@@ -41,9 +41,16 @@ public static IServiceCollection AddDatabase(this IServiceCollection services, I
 
         services.AddScoped<ITenantProvider>(sp =>
         {
-            var accountName = sp.GetService<IHttpContextAccessor>()?.HttpContext?.User
+            var context = sp.GetService<IHttpContextAccessor>()?.HttpContext;
+            var accountName = context?.User
                 .FindFirstValue(CustomClaimTypes.AccountName);
 
+
+            if (environment.IsDevelopment() && (context?.Request.Path == "/health/storage" || context?.Request.Path == "/ApplyDatabaseMigrations"))

```

Comment:
What I was thinking is we could have a environment conditional `MapGet("/")` so that when in development it injects `ITenantStorageFactory` and we just attempt to do something minor in the database. This was the application would launch to it and if there are database issues it will show the fancy exception page. 

But in non-dev it does what it does now.

Reply:
In that case we wouldn't need a seperate check for `Path == "/ApplyDatabaseMigrations"`

Reply:
@justindbaur Smart idea with the `/` path, but we would probably still need to whitelist `/ApplyDatabaseMigrations` because the button you click in the DatabaseExceptionHandler calls the `/ApplyDatabaseMigrations` -- are you implying we call `context.MigrateAsync` ourself?

Reply:
Resolved in a call, I wasn't aware `/ApplyDatabaseMigrations` was a post under the hood by the migrations exception page. We did decide to run it ourselves in root because we also want to seed the database for easier development anyways.